diff -Naur nginx-1.15.9/src/event/ngx_event_openssl.c nginx-1.15.9-new/src/event/ngx_event_openssl.c
--- a/src/event/ngx_event_openssl.c	2019-02-26 15:29:22.000000000 +0000
+++ b/src/event/ngx_event_openssl.c	2019-03-05 11:09:43.338466961 +0000
@@ -164,6 +164,18 @@
 
 #endif
 
+    // Enable FIPS mode for OpenSSL
+    int ret = FIPS_mode_set(1);
+    if (ret != 1) {
+        ERR_load_crypto_strings();
+        ERR_print_errors_fp(stderr);
+        
+        ngx_ssl_error(NGX_LOG_EMERG, log, 0, "FIPS_mode_set() failed: %lx.", ERR_get_error());
+        return NGX_ERROR;
+    } else {
+        fprintf(stderr,"*** IN FIPS MODE ***\n");
+    }
+
 #if OPENSSL_VERSION_NUMBER >= 0x0090800fL
 #ifndef SSL_OP_NO_COMPRESSION
     {
